src/core/cdr/src/dds_cdrstream.c

```diff
@@ -954,6 +955,7 @@ static uint32_t read_union_discriminant (dds_istream_t * __restrict is, uint32_t
       break;
     default: return 0;
   }
+  return 0;
```

Comment:
I may be nitpicking here: but I don't like changing *this* function in *this* way. The reason is the `default: return 0` just above it, and the fact that there is only one (dead-end) path through the function that runs into triggers this MISRA violation.

There are a number of ways it can be done that would make me happy (insofar as anything MISRA can make me happy):
* there is the obvious `default: abort (); return 0;` solution (line 953 in new numbers)
* or `default: break;` (line 956)
* or, recognise that the default case on line 956 is already an impossible one, and so instead make both default cases a simple `break`, and here do
```suggestion
  abort ();
  return 0;
```

@dpotman I *am* correct in that the default cases are both impossible, am I not?

Reply:
@eboasson yes, that's correct (and also also ensured in `normalize_uni_disc`), the IDL compiler or dynamic type builder is expected to only use one of the listed types as union discriminant.